### DISCUSSION TITLE
Load cpu every 15 minutes why?

### DISCUSSION THREAD

**Discussion by pbenaim** [photoprism/photoprism#3653] [Q&A] [answered]
With Zabbix on my server, I can see that Photoprism load the cpu every 15 minutes. 🤔

Normally Photoprism must be in idle state. ( no creating index, all convert are done, no request for web page, no WebDAV) 

I'm that this load come from Photoprism because when I stop it, no more load, so is not another process on my server... 

Any idea to correct this "issue"? 

![chart](https://github.com/photoprism/photoprism/assets/24461020/bc6f3fe9-b301-417a-852d-be9ccf8d196b)

**Comment by lastzero** [comment, marked as answer]
It's maintaining the index, eg improves face recognition performance, updates photo titles based on recognized persons, and estimates positions.

**Comment by pbenaim** [comment]
OK, thanks. 

So Photoprism stop to do that after one delay or is forever?

**Reply by lastzero** [reply]
We've optimized this in recent versions, however your question did not include any version information. Activity is generally kept to a minimum, depending on how many pictures you have. Since this is done by checking the database, you may see some minor CPU activity. Is this a huge problem?

**Reply by pbenaim** [reply]
Oups, my ver is 230719-73fa7bbe8, the last release. 

No, is not really a problem, is just a little bit strange to see activity when is supposed to be in idle state. 😏

**Comment by pbenaim** [comment]
Just a suggestion, maybe offer the option to disable this option 🤔

**Reply by lastzero** [reply]
https://docs.photoprism.app/getting-started/config-options/#index-workers

**Reply by fanyangto** [reply]
I am using the latest photoprism (PhotoPrism® CE Build 250707-d28b3101e).  This every 15 minute indexing looks still strange to me. Happens only a few hours per day, but not all day. Are you suggesting we should configure the index workers to zero as a way to disable this feature ? Would this cause problem as it disabling indexing entirely. 

The photo source has not changed for days. But the indexing still happens every day for a few hours. And each time only about 1000 photos processed. It looks like this:

time="2025-10-06T19:16:44Z" level=info msg="index: updated 996 photos [52.235230989s]"
time="2025-10-06T19:31:51Z" level=info msg="index: updated 1,000 photos [59.144285859s]"
time="2025-10-06T19:46:02Z" level=info msg="index: updated 272 photos [9.748673919s]"
time="2025-10-06T20:01:48Z" level=info msg="index: updated 959 photos [56.106591494s]"
time="2025-10-06T20:16:24Z" level=info msg="index: updated 801 photos [31.688731262s]"

On the other hand, I have another photoprism instance (same version) that never has this every 15 minute indexing behavior.

**Reply by omerdduran** [reply]
Hi. Don’t set index workers to 0—that doesn’t “disable” indexing and isn’t recommended; if you need to limit load, use PHOTOPRISM_INDEX_WORKERS=1. The 15-minute runs are the background worker waking on the default interval (PHOTOPRISM_WAKEUP_INTERVAL) and doing incremental index tasks whenever it sees metadatachanges or pending jobs, so you get batches of ~1k updates. To avoid scheduled rescans, leave PHOTOPRISM_INDEX_SCHEDULE empty; optionally raise PHOTOPRISM_WAKEUP_INTERVAL (e.g., 1h)

**Reply by fanyangto** [reply]
This helps. Thank you.

Just to clarify, PHOTOPRISM_INDEX_SCHEDULE  is the rescan of the source directory for changes, and PHOTOPRISM_WAKEUP_INTERVAL is for pending jobs/meta data changes ? Why every 15 minute does it update only about 1k photos? I couldn't find any documentation on what they are.

**Reply by omerdduran** [reply]
Yes. PHOTOPRISM_INDEX_SCHEDULE is the cron-like schedule for automatic library (re)index scans of the originals folder, while PHOTOPRISM_WAKEUP_INTERVAL controls how often the background worker wakes up to run face recognition, file sync, and metadata housekeeping (default 15 minutes). The ~1k “updated” lines you see are because those background/maintenance updates run in paged batches to keep database transactions small
